src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,233 @@
+
+    """
+    tokenizer_class = "AutoTokenizer"
+    attributes = ["tokenizer"]
+
+    preset_shape = {
+        "semantic_prompt": 1,
+        "coarse_prompt": 2,
+        "fine_prompt": 2,
+    }
+
+    def __init__(self, tokenizer, speaker_embeddings=None):
+        super().__init__(tokenizer)
+
+        self.speaker_embeddings = speaker_embeddings
+
+    @classmethod
+    def from_pretrained(
+        cls, pretrained_processor_name_or_path, speaker_embeddings_file_name="speaker_embeddings.npz", **kwargs
+    ):
+        r"""
+        Instantiate a Bark processor associated with a pretrained model.
+
+        Args:
+            pretrained_model_name_or_path (`str` or `os.PathLike`):
+                This can be either:
+
+                - a string, the *model id* of a pretrained [`BarkProcessor`] hosted inside a model repo on
+                  huggingface.co. Valid model ids can be located at the root-level, like `bert-base-uncased`, or
+                  namespaced under a user or organization name, like `dbmdz/bert-base-german-cased`.
+                - a path to a *directory* containing a processor saved using the [`~BarkProcessor.save_pretrained`]
+                  method, e.g., `./my_model_directory/`.
+            speaker_embeddings_file_name (`str`, *optional*, defaults to `"speaker_embeddings.npz"`):
+                The name of the `.npz` file containing the speaker_embeddings (e.g `"speaker_embeddings.npz"`) located
+                in `pretrained_model_name_or_path`. If `None`, no speaker_embeddings is loaded.
+            **kwargs
+                Additional keyword arguments passed along to both
+                [`~tokenization_utils_base.PreTrainedTokenizer.from_pretrained`].
+        """
+
+        if speaker_embeddings_file_name is not None:
+            speaker_embeddings = get_file_from_repo(
+                pretrained_processor_name_or_path,
+                speaker_embeddings_file_name,
+                subfolder=kwargs.pop("subfolder", None),
+                cache_dir=kwargs.pop("cache_dir", None),
+                force_download=kwargs.pop("force_download", False),
+                proxies=kwargs.pop("proxies", None),
+                resume_download=kwargs.pop("resume_download", False),
+                local_files_only=kwargs.pop("local_files_only", False),
+                use_auth_token=kwargs.pop("use_auth_token", None),
+                revision=kwargs.pop("revision", None),
+            )
+            if speaker_embeddings is None:
+                logger.warning(
+                    f"""`{os.path.join(pretrained_processor_name_or_path,speaker_embeddings_file_name)}` does not exists
+                    , no preloaded speaker embeddings will be used - Make sure to provide a correct path if wanted,
+                    otherwise set `speaker_embeddings_file_name=None`."""
+                )
+            else:
+                speaker_embeddings = np.load(speaker_embeddings, allow_pickle=True)
+
```

Comment:
I had to use `pickle` here, but I'm not sure about the safety of the method. I don't see another way to save/load a nested dictionary of `np.ndarray` tbh.

WDTY?

Reply:
First thought when looking at this is that all of the speaker embeddings have to be downloaded at once. Is it likely that a user will use all of these presets in a single inference / training session? 

I would instead structure it like so: 

In the model's repo, we have a folder `speaker_embeddings`, similar to the one you currently have. Within this folder, we could do 1 of 2 things: 

1. Save each of the arrays individually with a suffix e.g. `de_speaker_0_semantic_prompt.npy`
2. Have a nested folder structure i.e. `MODEL_REPO_NAME/bark-xxx/speaker_embeddings/de_speaker_0/semantic_prompt.py`

Personally, I'd go for 1. 

Then at the top level of the model repo, we have a file `speaker_embedding_paths.json`, which is a nested dict, listing where all the files are: 

```json
{
    "de_speaker_0": {
        "semantic_prompt": "MODEL_REPO/bark-xxx/speaker_embeddings/de_pseaker_0_semantic_prompt.npy",
        "coarse_prompt":  "MODEL_REPO/bark-xxx/speaker_embeddings/de_pseaker_0_coarse_prompt.npy",
        "fine_prompt": "MODEL_REPO/bark-xxx/speaker_embeddings/de_pseaker_0_fine_prompt.npy",
    },
}
```

or possibly shorter oaths just with the filename. I'd go for the full repo path, as that means others can point and use the embeddings in other repos, without having to copy. 

When someone requests a speaker embedding, we can first check if it's downloaded, if it is, load from there, otherwise we download from the path specified in the speaker_embeddings dict

Reply:
Hi @amyeroberts, 
I really like 1, I think it's the best way to be both safe and consistent with the original Bark `speaker_embeddings` nested dictionary. I will do it in the following hours. Thanks for your insights!